turtlebot3_fake_node/src/turtlebot3_fake_node.cpp

```diff
@@ -0,0 +1,266 @@
+
+  // Initialise subscribers
+  cmd_vel_sub_ = this->create_subscription<geometry_msgs::msg::Twist>(
+    "cmd_vel", qos, std::bind(&Turtlebot3Fake::command_velocity_callback, this, std::placeholders::_1));
+
+  /************************************************************
+  ** Start update thread
+  ************************************************************/
+  update_timer_ = this->create_wall_timer(30ms, std::bind(&Turtlebot3Fake::update_callback, this));
+
+  RCLCPP_INFO(this->get_logger(), "Turtlebot3 Fake Node Initialised");
+}
+
+Turtlebot3Fake::~Turtlebot3Fake() 
+{  
+  RCLCPP_INFO(this->get_logger(), "Turtlebot3 Fake Node Terminated");
+}
+
+/********************************************************************************
+** Init functions
+********************************************************************************/
+void Turtlebot3Fake::init_parameters()
+{
+  // Declare parameters that may be set on this node
+  this->declare_parameter("joint_states_frame");
+  this->declare_parameter("odom_frame");
+  this->declare_parameter("base_frame");
+  this->declare_parameter("wheels.separation");
+  this->declare_parameter("wheels.radius");
+
+  // Get parameters from yaml
+  this->get_parameter_or<std::string>("joint_states_frame", joint_states_.header.frame_id, "base_footprint");
+  this->get_parameter_or<std::string>("odom_frame", odom_.header.frame_id, "odom");
+  this->get_parameter_or<std::string>("base_frame", odom_.child_frame_id, "base_footprint");
+  this->get_parameter_or<double>("wheels.separation", wheel_seperation_, 0.0);
+  this->get_parameter_or<double>("wheels.radius", wheel_radius_, 0.0);
+}
+
+void Turtlebot3Fake::init_variables()
+{
+  // Initialise variables
+  wheel_speed_cmd_[LEFT]  = 0.0;
+  wheel_speed_cmd_[RIGHT] = 0.0;
+  goal_linear_velocity_   = 0.0;
+  goal_angular_velocity_  = 0.0;
+  cmd_vel_timeout_        = 1.0;
+  last_position_[LEFT]    = 0.0;
+  last_position_[RIGHT]   = 0.0;
+  last_velocity_[LEFT]    = 0.0;
+  last_velocity_[RIGHT]   = 0.0;
+
+  // TODO: Find more accurate covariance
+  // double pcov[36] = { 0.1,   0,   0,   0,   0, 0,
+  //                       0, 0.1,   0,   0,   0, 0,
+  //                       0,   0, 1e6,   0,   0, 0,
+  //                       0,   0,   0, 1e6,   0, 0,
+  //                       0,   0,   0,   0, 1e6, 0,
+  //                       0,   0,   0,   0,   0, 0.2};
+  // memcpy(&(odom_.pose.covariance), pcov, sizeof(double)*36);
+  // memcpy(&(odom_.twist.covariance), pcov, sizeof(double)*36);
+
+  odom_pose_[0] = 0.0;
+  odom_pose_[1] = 0.0;
+  odom_pose_[2] = 0.0;
+  odom_vel_[0] = 0.0;
+  odom_vel_[1] = 0.0;
+  odom_vel_[2] = 0.0;
+
+  joint_states_.name.push_back("wheel_left_joint");
+  joint_states_.name.push_back("wheel_right_joint");
+  joint_states_.position.resize(2, 0.0);
+  joint_states_.velocity.resize(2, 0.0);
+  joint_states_.effort.resize(2, 0.0);
+
+  rclcpp::Clock clock(RCL_SYSTEM_TIME);
```

Comment:
You can use `now()` function to get node time.
http://docs.ros2.org/dashing/api/rclcpp/classrclcpp_1_1Node.html#af706b231620f1c120b7ccd738ec31867

Reply:
Please check it.

Reply:
now() function is used in the very next line to update the variable 'prev_update_time_'. I hope that is what he meant? @routiful

Reply:
@rjshim I suggest that you can use `this->now()` function to get time of this node.